src/modules/documents/documents.controller.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Controller,
+  Post,
+  Body,
+  UseInterceptors,
+  UploadedFile,
+} from '@nestjs/common';
+import { DocumentsService } from './documents.service';
+import { CreateDocumentBySubjectDto } from './dto/create-document.dto';
+import { Auth } from '@/common/decorators/auth.decorator';
+import { AuthType } from '@/common/enums/auth-type.enum';
+import { FileInterceptor } from '@nestjs/platform-express';
+import {
+  DOCUMENT_FILE_NAME,
+  DOCUMENT_FILE_PATH,
+  DOCUMENT_FILE_SIZE,
+  DOCUMENT_FILE_TYPE,
+} from '@/common/constants/documents.constant';
+import { setUploadOptions } from '@/common/config/multer.config';
+import { AuthUser } from '@/common/decorators/auth-user.decorator';
+import { User } from '@prisma/client';
+
+@Controller('api/v1/documents')
+export class DocumentsController {
+  constructor(private readonly documentsService: DocumentsService) {}
+
+  @Post()
+  @UseInterceptors(
+    FileInterceptor(
+      DOCUMENT_FILE_NAME,
+      setUploadOptions(
+        DOCUMENT_FILE_PATH,
+        DOCUMENT_FILE_TYPE,
+        DOCUMENT_FILE_SIZE,
+      ),
+    ),
+  )
+  createDocumentBySubject(
+    @AuthUser() user: User,
+    @UploadedFile() file: Express.Multer.File,
+    @Body()
+    createDocumentBySubjectDto: CreateDocumentBySubjectDto,
+  ) {
+    return this.documentsService.createDocumentBySubject(
+      user,
+      file.path,
+      createDocumentBySubjectDto,
+    );
```

Comment:
sửa lại dto với tên hàm luôn em nhé
```suggestion
  createCourseDocument(
    @AuthUser() user: User,
    @UploadedFile() file: Express.Multer.File,
    @Body()
    createCourseDocumentDto: CreateCourseDocumentDto,
  ) {
    return this.documentsService.createCourseDocument(
      user,
      file.path,
      createCourseDocumentDto,
    );
```